app/javascript/packages/verify-flow/services/api.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Submits the given payload to the API route controller associated with the current path, resolving
+ * to a promise containing the parsed response JSON object.
+ *
+ * @param payload Payload object.
+ *
+ * @return Parsed response JSON object.
+ */
+export async function post<Response = any>(payload: object): Promise<Response> {
```

Comment:
Yeah those options make sense too!

Reply:
Updated to `json` and `csrf` options in 7b42cf64b.

I'm a bit concerned about usability impact, since while these options make sense for a generic API POST helper, the intended use-case here is pretty specific, so I'd be inclined to optimize it to "just work" as much as possible. A function rename may ultimately be the better option, but I also don't want it to be too wordy either 🤷